bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/handler/NikobusPcLinkHandler.java

```diff
@@ -159,9 +160,20 @@ private void processReceivedValue(byte value) {
 
     @SuppressWarnings("null")
     public void addListener(String command, Runnable listener) {
+        logger.trace("adding listener for '{}'", command);
         if (commandListeners.put(command, listener) != null) {
             logger.warn("Multiple registrations for '{}'", command);
         }
+        
+        // the connection must be open to be able to receive messages from the bus
+        try {
+            connectIfNeeded(connection);
```

Comment:
Hm is this really required here? connection will be establish when first command is send (`REFRESH` for example) ... so I guess not need to call it here too, or did I miss something?

Reply:
it is needed, when the binding starts up the connection is not opened and therefore it is not reading bus messages coming from physical buttons.

Reply:
When binding starts up, it will schedule `refresh` in it's `initialize` function - and that will establish the connection. If that would not be the case then binding would not work at all, right?

The only scenario I can think of that connection would not be established is if there are no modules defined for a bridge - than binding will not try to connect since there is no Nikobus module configured. Are you using a configuration without any Nikobus module (switch/dimmer/rollershutter)?

Reply:
I am slowly moving a way from Nikobus. My modules are failing one by one and soon there will be no spare parts. But I want to keep the physical buttons in my house. So I would like the binding to work even without any modules defined.

Reply:
That is not good to hear, hope my modules will last a bit longer :)

In that case (setup without modules) I would suggest to move this `connect` related code to `refresh` - please note `addListener` will be most probably called a lot of times and no real need to request connection here.

So I would suggest to modify 

```
if (things.isEmpty()) {
            logger.debug("Nothing to refresh");
            return;
        }
```

within the `refresh` method:

```
if (things.isEmpty()) {
  if (!commandListeners.isEmpty()) {
       // the connection must be open to be able to receive messages from the bus
        try {
            connectIfNeeded(connection);
         } catch (IOException e) {
            logger.debug("Sending command failed due {}", e.getMessage(), e);
            connection.close();
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
  }
 logger.debug("Nothing to refresh");
 return;
 }
```

Makes sense?